architecture/2024-10-deploymenttemplate-controller.md

```diff
@@ -0,0 +1,421 @@
+	Az          *Az          `json:"az,omitempty"`
+	AWS         *AWS         `json:"aws,omitempty"`
+	Deployments *Deployments `json:"deployments,omitempty"`
+}
+
+type Radius struct {
+	Type  string `json:"type,omitempty"`
+	Value Value  `json:"value,omitempty"`
+}
+
+type Value struct {
+	Scope string `json:"scope,omitempty"`
+}
+
+// ...
+
+// DeploymentTemplateStatus defines the observed state of the Deployment Template.
+type DeploymentTemplateStatus struct {
+  // ObservedGeneration is the most recent generation observed for this DeploymentTemplate.
+  ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+  // Template is the ARM JSON manifest that defines the resources to deploy.
+  Template string `json:"template"`
+
+  // Parameters is the ARM JSON parameters for the template.
+  Parameters string `json:"parameters"`
+  
+  // Scope is the resource id of the Radius scope.
+  Scope string `json:"scope,omitempty"`
+
+  // Resource is the resource id of the deployment.
+  Resource string `json:"resource,omitempty"`
+
+  // Operation tracks the status of an in-progress provisioning operation.
+  Operation *ResourceOperation `json:"operation,omitempty"`
+
+  // Phrase indicates the current status of the Deployment Template.
+  Phrase DeploymentTemplatePhrase `json:"phrase,omitempty"`
+
+  // Message is a human-readable description of the status of the Deployment Template.
+  Message string `json:"message,omitempty"`
+}
+
+// DeploymentTemplatePhrase is a string representation of the current status of a Deployment Template.
+type DeploymentTemplatePhrase string
+
+const (
+  // DeploymentTemplatePhraseUpdating indicates that the Deployment Template is being updated.
+  DeploymentTemplatePhraseUpdating DeploymentTemplatePhrase = "Updating"
+
+  // DeploymentTemplatePhraseReady indicates that the Deployment Template is ready.
+  DeploymentTemplatePhraseReady DeploymentTemplatePhrase = "Ready"
+
+  // DeploymentTemplatePhraseFailed indicates that the Deployment Template has failed.
+  DeploymentTemplatePhraseFailed DeploymentTemplatePhrase = "Failed"
+
+  // DeploymentTemplatePhraseDeleting indicates that the Deployment Template is being deleted.
+  DeploymentTemplatePhraseDeleting DeploymentTemplatePhrase = "Deleting"
+
+  // DeploymentTemplatePhraseDeleted indicates that the Deployment Template has been deleted.
+  DeploymentTemplatePhraseDeleted DeploymentTemplatePhrase = "Deleted"
+)
+```
+
+#### DeploymentResource Custom Resource Definition
+
+The `DeploymentResource` CRD is another CRD that will be responsible for tracking the state of individual resources. The CRD will have the following fields:
+
+```go
+// DeploymentResourceSpec defines the desired state of a Deployment Resource.
+type DeploymentResourceSpec struct {
+  // Scope is the resource ID of the scope.
+	Scope string `json:"scope,omitempty"`
+
+  // ResourceId is the Radius resource Id.
+  ResourceId string `json:"resourceId"`
```

Comment:
could we call this `ID`? We usually just say `id` where a resource ID appears in code. Resource ID doesn't add additional clarity.